### ISSUE TITLE
Improve engine's control UI by generating customized TTL files

### ISSUE THREAD

**Issue by riban-bw** [zynthian/zynthian-issue-tracking#706]
**Is your feature request related to a problem? Please describe.**
There are many controller titles that do not fit the space assigned to display the controller title. Helm is particularly poor but other engines suffer this issue. Automatic truncation and line split can result in undesirable effects. Helm's controls all have words separated by underscores. This looks ugly and is poor UX.

**Describe the solution you'd like**
Each parameter should have a suitable title that accurately and concisely describes the function. They should have spaces between words to allow automatic line split to operate properly. Length should be limited to reduce risk of poor display.

**Additional context**
Can this be done with a custom ttl?

**Comment by jofemodo** [comment]
Yes. It can be done with a custom TTL. In fact, it's been done for a number of engines: Dexed, OBXd, NoizeMaker, etc.
The process is simple and it's documented in the forum:

+ https://discourse.zynthian.org/t/lv2-controllers-groups/6201
+ https://discourse.zynthian.org/t/noisemaker-with-grouped-parameters/6584
+ https://discourse.zynthian.org/t/dexed-with-grouped-parameters/6588

BTW, parameters can be sorted and grouped, what grately improves engine control and UX.

**Comment by jofemodo** [comment]
We should relay on the community for this kind of tasks, but some effort is needed for managing the crew's workforce ;-)

**Comment by jofemodo** [comment]
To complete the task:

Generating engine customized TTLs allows to make several iimprovements in the engine's control UI:

+ **Rename** control paramaters
+ **Group** control parameters
+ Adjust **value range** & **add ticks/labels** to control parameters

**Comment by riban-bw** [comment]
I have done Helm, adding groups, ordering controls and setting correct control type, e.g. enumerated lists as required.

I also added two new options that allow ordering (sorting) the display of controls and hiding controls in the UI.

If a LV2 port is flagged with <http://lv2plug.in/ns/ext/port-props#notOnGUI> then it will be hidden. A common example of this is 'lv2_freewheel'. For example:
```
    lv2:port [
        a lv2:InputPort, lv2:ControlPort ;
        lv2:index 1 ;
        lv2:symbol "lv2_freewheel" ;
        lv2:name "Freewheel" ;
        lv2:default 0.0 ;
        lv2:minimum 0.0 ;
        lv2:maximum 1.0 ;
        lv2:designation <http://lv2plug.in/ns/lv2core#freeWheeling> ;
        lv2:portProperty lv2:toggled, <http://lv2plug.in/ns/ext/port-props#notOnGUI> ;
    ] ;
```

If a port has the parameter lv2:displayPriority then this is used as a hint to the order in which to display controls. The higher the value the earlier the control is displayed. (That is a bit awkward but is the LV2 spec.) Controls with the same display priority will be ordered by their index. By default controls get a priority of zero, i.e. they appear last.

Missing options that would be beneficial include:

- Ability to scale values, e.g. a port with range 0.0 to 1.0 should be able to be displayed with a range 0% to 100%.
- Ability to offset values, e.g. a port with range 0.0 to 1.0 should be able to be displayed with a range -0.5 to +0.5.

These two options would enable most of the outstanding issues with the Helm configuration and probably may other engines.

**Comment by jofemodo** [comment]
I have some nice ideas for the future. For instance, do you know about parameter groups?

http://lv2plug.in/ns/ext/parameters

I envision GUI widgets that could be auto-associated to parameter groups of specific types. I.e. nice GUI widgets for EnvelopeControls, FilterControls, etc. It would be super-cool!!
